src/libstd/sys/unix/args.rs

```diff
@@ -137,9 +138,11 @@ mod imp {
     fn clone() -> Vec<OsString> {
         unsafe {
             let _guard = LOCK.lock();
-            (0..ARGC)
+            (0..ARGC.load(Ordering::Relaxed))
                 .map(|i| {
-                    let cstr = CStr::from_ptr(*ARGV.offset(i) as *const libc::c_char);
+                    let cstr = CStr::from_ptr(
+                        *ARGV.load(Ordering::Relaxed).offset(i) as *const libc::c_char
```

Comment:
Would it make a difference to move this load out of the loop?

Reply:
Unlikely to make a perf difference, but I like it anyway, because it'll make it clear that there is no expectation of this loop to ever load a different value while looping.

Reply:
Fixed!